### PR TITLE
[Backport 1.11] DCOS-21305: Respect DC/OS min required version when installing a package.

### PR DESCRIPTION
**Pull request by juliangieseke**
Backport of https://github.com/dcos/dcos-ui/pull/2786

### CODE REVIEW

File: src/js/pages/catalog/PackageDetailTab.js [hunk 1]
```diff
@@ -9,18 +9,20 @@ import { StoreMixin } from "mesosphere-shared-reactjs";
...
+  from "#PLUGINS/services/src/img/icon-service-default-large@2x.png";
+import Image from "#SRC/js/components/Image";
+import ImageViewer from "#SRC/js/components/ImageViewer";
+import Loader from "#SRC/js/components/Loader";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+import Page from "#SRC/js/components/Page";
+import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
+import StringUtil from "#SRC/js/utils/StringUtil";
+
+const semver = require("semver");
```
Comment (DanielMSchmidt) [inline src/js/pages/catalog/PackageDetailTab.js]: I think import would be better and more consistent (for that reason also on the top)
Sorry I should have caught this in the review process, I think we need to leave it there for now, right?

File: tests/_fixtures/dcos/dcos-version.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "1.6",
+  "version": "1.11",
```
Comment (DanielMSchmidt) [inline tests/_fixtures/dcos/dcos-version.json]: 👍

File: src/js/pages/catalog/PackageDetailTab.js (hunk 1 above)
Comment (Fabs) [reply src/js/pages/catalog/PackageDetailTab.js]: I agree that we should have catched that on the normal PR. Now that it made in I would leave it :/

File: tests/pages/catalog/packages/PackagesTab-cy.js [hunk 3]
```diff
@@ -165,4 +165,24 @@ describe("Packages Tab", function() {
...
+      cy
+        .route(/dcos-version/, {
+          version: "1.6",
+          "dcos-image-commit": "null",
+          "bootstrap-id": "null"
+        })
+        .visitUrl({ url: "/catalog", logIn: true });
+    });
+
+    it("cant install package because of DCOS version", function() {
```
Comment (Fabs) [inline tests/pages/catalog/packages/PackagesTab-cy.js]: I am assuming somewhere else there is a test for the button being enabled. If not that would be nice to have the other case tested.

Also this logic here has quite some possible inputs. 
```
+      MetadataStore.version &&
+      cosmosPackage.minDcosReleaseVersion &&
+      semver.compare(
+        semver.coerce(MetadataStore.version),
+        semver.coerce(cosmosPackage.minDcosReleaseVersion)
+      ) < 0
```

I would have unit tested that logic.

**Comment by orlandohohmeier** [comment]
LGTM! 🚢

File: src/js/pages/catalog/PackageDetailTab.js (hunk 1 above)
Comment (orlandohohmeier) [reply src/js/pages/catalog/PackageDetailTab.js]: @juliangieseke could you pleas file a JIRA for this one?

**Pull request merged**